### ISSUE TITLE
bug: aria-label on ion-back-button is not propagated to native button

### ISSUE THREAD

**Issue by Abrissirba** [ionic-team/ionic-framework#24012]
### Prequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [X] v5.x
- [ ] v6.x

### Current Behavior

When having a ion-back-button without a text, the screen reader always says back even though I set aria-label to something else. 

### Expected Behavior

When setting an explicit aria-label, that label should be used instead of using a hardocded 'back'.

This becomes a problem if you don't use the text attribute and the app is used in other languages than english.

### Steps to Reproduce

Add an ion-back-button element withou text and set aria-label="custom back button label"

the hardcoded value is set here: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/back-button/back-button.tsx#L142

### Ionic Info

Ionic:

   Ionic CLI                     : 6.16.3 (C:\Users\marabr\AppData\Roaming\nvm\v12.17.0\node_modules\@ionic\cli)
   Ionic Framework               : @ionic/angular 5.3.1 (C:\Projects\SXBoendeApp\AppMonoRepo\node_modules\@ionic\angular)
   @angular-devkit/build-angular : 0.1000.8 (C:\Projects\SXBoendeApp\AppMonoRepo\node_modules\@angular-devkit\build-angular)
   @angular-devkit/schematics    : 10.2.0 (C:\Projects\SXBoendeApp\AppMonoRepo\node_modules\@angular-devkit\schematics)
   @angular/cli                  : 10.0.7 (C:\Projects\SXBoendeApp\AppMonoRepo\node_modules\@angular\cli)
   @ionic/angular-toolkit        : 2.3.3 (C:\Projects\SXBoendeApp\AppMonoRepo\node_modules\@ionic\angular-toolkit)

Cordova:

   Cordova CLI       : 10.0.0
   Cordova Platforms : not available
   Cordova Plugins   : not available

Utility:

   cordova-res : not installed globally
   native-run  : 1.4.1

System:

   NodeJS : v12.17.0 (C:\Program Files\nodejs\node.exe)
   npm    : 6.14.4
   OS     : Windows 10

**Comment by Abrissirba** [comment]
Thanks for the quick fix :)
